addon/modifiers/sortable-item.js

```diff
@@ -755,10 +755,26 @@ export default class SortableItemModifier extends Modifier {
     this.addEventListener();
     this.element.dataset.sortableItem=true;
     this.element.sortableItem = this;
+
+    // Find my group
+    // we have to schedule because the items are installed before the group
+    run.schedule('afterRender', () => {
+      let parent = this.element.parentElement;
+      while (parent) {
+        if (parent.sortableGroup) {
+          parent.sortableGroup.registerItem(this);
+          this.sortableGroup = parent.sortableGroup;
+          break;
+        }
+        parent = parent.parentElement;
+      }
+    });
```

Comment:
Ah, this makes me kind of sad.
Ember-sortable was originally implemented as separate entities that did not know  about each other without explicit metadata to track, e.g `sortable-group` and `sortable-item`. The ability to create higher order components via `yield` was on the major improvements done in `2.x` because it allowed us to get rid of most of the metadata needed to always keep the two connected. In almost all cases, you are using `sortable-group` and `sortable-item` together, so the tight coupling was warranted.

I wonder if there's a way we can do the same for `modifiers` in a higher order fashion.

Reply:
There was talk of a `modifier` helper similar to the `component` helper which would have definitely solved the concern here. Sadly that helper never came to be.

The main problem with the components was it as a fixed DOM element. Only changeable by tagName (going away with glimmer) and not generically useable by other addons components. Any element should be sortable, even elements of other components. Say there is another addon that does a really great job of ol/li management. We can not user ember-sortable with that add on because the sortable components insist on managing the DOM. . But with modifiers,  we can apply the modifiers to the other addons components and now those components are also sortable (This does require the other addon to use `...arguments` in the correct place. An easy fix and almost a guarantee when converted to glimmer).

Without a `modifier` helper, the only way to make modifiers exhibit the shared state is to require the consumer to assign the state on each modifier, a burden I thought needless when it could be derived for the consumer.

Reply:
Oh, FYI the above code exists in the group of the already approve code that is in the beta. This is just moving the code from the group to the item to support the async load.

Reply:
I agree that modifier does bring a whole new perspective into the ecosystem. We will need to re-visit the architecture and design.
But as to unblock current tester for modifiers, are there tests that we can write to ensure that this `async` behavior is resolved?

Reply:
https://github.com/st-h created a test in his PR https://github.com/adopted-ember-addons/ember-sortable/pull/354 which was closed in favor of this one. I copied that test over and its in this PR. It added items to the array after render and checks that the elements are added on screen and are then sortable

Reply:
sounds good.